atomic_reactor/plugins/pre_pyrpkg_fetch_artefacts.py

```diff
@@ -10,13 +10,41 @@
 
 This plugin should do it.
 """
-import os
+import re
 import subprocess
+from pathlib import Path
+from typing import Iterable, List
 
+from atomic_reactor.dirs import BuildDir
 from atomic_reactor.plugin import PreBuildPlugin
 from atomic_reactor.constants import PLUGIN_DISTGIT_FETCH_KEY
 
 
+MD5_CHECKSUM_RE = r'^[a-f0-9]+ [* ](.+)\n?$'
+BSD_CHECKSUM_RE = r'^[^ ]+ \((.+)\) = [a-f0-9]+\n?$'
+
+
+def read_filenames_from_sources(sources_file: Path) -> Iterable[str]:
+    """Read and return the file names listed in the sources file.
+
+    :param sources_file: the file name of the sources file.
+    :type sources_file: pathlib.Path
+    :return: a iterator of file names.
+    :rtype: iterable[str]
+    """
+    known_patterns = (re.compile(BSD_CHECKSUM_RE), re.compile(MD5_CHECKSUM_RE))
```

Comment:
Is the sources file format documented somewhere?

Reply:
yeah that would be useful to know

Reply:
AFAIK, no such document. I asked the rpkg maintainer, let's see if there is any news.